src/main/java/uk/gov/hmcts/reform/cpo/security/ServiceAuthorizationConfig.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.hmcts.reform.cpo.security;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@Validated
+@ConfigurationProperties(prefix = "s2s")
+public class ServiceAuthorizationConfig {
+
+    private Map<String, String> permissions;
+
+    public boolean hasPermissions(String serviceId, Permission crudPermission) {
+        return permissions.containsKey(serviceId) && permissions.get(serviceId).contains(crudPermission.getLabel());
+    }
+
+    public void setAuthorizations(Map<String, @Valid ServicePermissionInfo> authorizations) {
+        permissions = new HashMap<>();
+        authorizations
```

Comment:
```suggestion
       permissions = authorizations.values().stream()
            .collect(Collectors.toMap(ServicePermissionInfo::getId, ServicePermissionInfo::getPermission));
```

Reply:
👍

Reply:
Fixed in https://github.com/hmcts/cpo-case-payment-orders-api/pull/68/commits/970b0c16e75dc6be385b97fa1c0dd42e08d20d2b